edalize/cocotb.py

```diff
@@ -0,0 +1,162 @@
+import logging
+import os
+import pathlib
+
+import cocotb
+
+from edalize import get_edatool
+
+logger = logging.getLogger(__name__)
+
+
+# This class could be replaced by calls to cocotb-config. Knowledge of these
+# simulator arguments belongs in the cocotb codebase.
+class CocotbConfig:
+
+    @staticmethod
+    def get_simulator_args(simulator, toplevel_lang):
+        if simulator == 'ghdl':
+            return CocotbConfig.get_ghdl_args(toplevel_lang)
+        elif simulator == 'icarus':
+            return CocotbConfig.get_icarus_args(toplevel_lang)
+        else:
+            raise RuntimeError('Simulator ' + simulator + ' is not supported by cocotb backend')
+
+    @staticmethod
+    def get_ghdl_args(toplevel_lang):
+        if toplevel_lang != 'vhdl':
+            raise RuntimeError('GHDL only supports VHDL')
+
+        ghdl_cocotb_lib = pathlib.Path(os.path.dirname(cocotb.__file__)) / 'libs' / 'libcocotbvpi_ghdl.so'
+        args = f"--vpi={ghdl_cocotb_lib}"
+
+        return args
+
+    @staticmethod
+    def get_icarus_args(toplevel_lang):
+        if toplevel_lang != 'verilog':
+            raise RuntimeError('Icarus only supports Verilog')
+
+        cocotb_lib_dir = pathlib.Path(os.path.dirname(cocotb.__file__)) / 'libs'
+        args = f"-M {cocotb_lib_dir} -m libcocotbvpi_icarus"
+
+        return args
```

Comment:
Returning a string seems foolish here. Can you return a list of args instead?

```suggestion
        return ["-M", cocotb_lib_dir, "-m", "libcocotbvpi_icarus"]
```

Reply:
Yes, it should be a list. I've written this bit several times and haven't gone back over it for sanity!

Thanks.